### PR TITLE
Feature / validations and model specs

### PR DESCRIPTION
**Pull request by John-Kibirige**
## Activities worked on in this milestone 
- adding validations to the post and user models 
- creating tests for all the four models 
- fixing linter errors

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  it 'likes counter should be invalid if its -1' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'posts count should return 1' do
+    expect(subject.author.posts_counter).to be_truthy
+  end
+end
```
Comment (fobadara) [inline spec/models/post_spec.rb:60]: [REQUIRED] I can't find a spec for the custom method `update_post_counter`

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  subject { Comment.new(text: 'comment test', author_id: @user.id, post_id: @post.id) }
+
+  before { subject.save }
+
+  it 'comments shoulld be increment by 1' do
+    expect(subject.post.comments_counter).to eq(1)
+  end
+end
```
Comment (fobadara) [inline spec/models/comment_spec.rb]: [REQUIRED] This shows that we can create a comment but I can't find a spec for your custom method `update_comments_count`

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  subject { Like.new(author_id: @user.id, post_id: @post.id) }
+
+  before { subject.save }
+
+  it 'likes shoulld be increment by 1' do
+    expect(subject.post.likes_counter).to eq(1)
+  end
+end
```
Comment (fobadara) [inline spec/models/like_spec.rb:21]: [REQUIRED] I can't also find the test for updating likes count

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @John-Kibirige, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Project looks professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @John-Kibirige , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**